File: demos/edit_modes/scripts/playground.js
solver = new c.SimplexSolver();
var min = new c.Variable({
  value: 10
});
var max = new c.Variable({
  value: 15
});


solver.addStay(min);
solver.addStay(max,c.Strength.medium);

/*
solver.addConstraint(new c.StayConstraint(min));
solver.addConstraint(new c.StayConstraint(max));
*/

var expr = c.Expression.fromVariable(min,solver).plus(new c.Expression(20));
var constr = new c.Equation(max, expr);

//var constr = new c.Inequality(min, c.LEQ, max, c.Strength.required);
solver.addConstraint(constr);

console.log("min : "+min.value);
console.log("max : "+max.value);

solver.addEditVar(min)
        .beginEdit();

solver.suggestValue(min, 40);
solver.resolve();

console.log("min : "+min.value);
console.log("max : "+max.value);

File: demos/kiwi/examples/playground.cpp
#include <kiwi.h>

using namespace kiwi;

int main( int argc, const char* argv[] )
{
    Solver solver{};

    Variable t1("t1");
    Variable t2("t2");
    Variable min1("min1");
    Variable max1("max1");
    Variable nom1("nom1");

    solver.addEditVariable(t1, strength::strong);
    solver.addEditVariable(t2, strength::strong);
    solver.addEditVariable(min1, strength::required);
    solver.addEditVariable(max1, strength::required);
    solver.addEditVariable(nom1, strength::medium);

    solver.addConstraint(min1 >= 0);
    solver.addConstraint(min1 <= nom1);
    solver.addConstraint(nom1 <= max1);
    solver.addConstraint(t2 == t1 + nom1);

    solver.dump();

    return 0;
}

